static/js/components/search/CustomSortingSelect.js

```diff
@@ -42,15 +41,10 @@ export default class CustomSortingSelect extends Select {
           Sort by:
         </span>
         <select onChange={this.onChange} value={this.getSelectedValue()} ref="sortSelectField">
-          {_.map(items, ({key, label, title, disabled, docCount}) => {
+          {_.map(items, (item) => {
             return (
-              <option key={key} value={key} disabled={disabled}>
-                {this.optionText(
-                  translate(label || title || key),
-                  showCount,
-                  docCount,
-                  countFormatter
-                )}
+              <option key={item.key} value={item.key} disabled={item.disabled}>
+                {this.optionText(translate, showCount, countFormatter, item)}
```

Comment:
ok, sorry maybe my last comment wasn't so clear. I was suggesting something like this:

```js
options (
  items,
  translate,
  showCount,
  countFormatter,
) {
  return items.map(item => {
    let text = translate(item.label || item.title || item.key);
    if (showCount && item.docCount !== undefined) {
      text += ` (${countFormatter(item.docCount)})`;
    }
    
    return <option key={item.key} value={item.key} disable={item.disabled}>
      { text }
    </option>;
  });
}
```

then in the JSX you'd just call

```js
{ this.options(items, translate, showCount, countFormatter) }
```

does that make sense?